llmfoundry/data/finetuning/dataloader.py

```diff
@@ -145,6 +148,21 @@ def build_finetuning_dataloader(cfg: DictConfig, tokenizer: Tokenizer,
         )
 
     else:
+        if '://' in cfg.dataset.hf_name:
+            with tempfile.TemporaryDirectory() as tmp_dir:
+                supported_extensions = ['jsonl', 'csv', 'parquet']
+                for extension in supported_extensions:
+                    name = f'{cfg.dataset.hf_name}/{cfg.dataset.split}.{extension}'
+                    location = f'{tmp_dir}/{name}'
+                    try:
+                        get_file(cfg.dataset.hf_name, location)
+                    except FileNotFoundError as e:
+                        if extension == supported_extensions[-1]:
+                            raise e
+                        continue
+                    cfg.dataset.hf_name = extension
+                    cfg.dataset.hf_kwargs['data_dir'] = tmp_dir
+                    cfg.dataset.hf_kwargs['split'] = cfg.dataset.split
```

Comment:
When we call `dataset = dataset_constructor.build_from_hf(cfg.dataset, tokenizer)` the `build_from_hf` method will set the `split` kwarg based on `cfg.dataset.split`, so I think you do not need this line.